can/io/blf.py

```diff
@@ -370,6 +367,7 @@ def __init__(
         append: bool = False,
         channel: int = 1,
         compression_level: int = -1,
+        **options: Any,
```

Comment:
Added `**options` to BLFWriter API.

Reply:
Assuming that `compression_level` is specific to the BLFWriter, perhaps it should be moved into the `options` dict as well?

Reply:
Arguments specific to a `Writer` could be made more accessible if able to be set by `**options` . Currently, the only way to set `compression_level` would be to bypass the `Logger` (or `RollingLogger`) object and call the `BLFWriter` object itself. 

When going about these changes, I'll want to make sure that the API is either unaffected by this improvement, or that a deprecation warning is added (similar to what I did [here](https://github.com/j-c-cook/python-can/blob/5c232483e97d92b002886ada109f4f6500ffec25/can/io/logger.py#L433-L439) for `SizedRotatingLogger`).

Reply:
It is specific to the BlfWriter (and soon Mf4Writer) but i think it's more readable when it is not hidden away in `**options`

Reply:
I have confirmed that `compression_level` can be modified by passing via the argument in `**kwargs`. @zariiii9003's CLI arguments https://github.com/zariiii9003/python-can/tree/extra_args2 will provide the ability to modify variables such as `compression_level`. For example:
```
python -m can.logger -i socketcan -c vcan0 -b 250000 -f file.blf -s 100000 --compression-level=1 --max-container-size=200000
```
Once https://github.com/hardbyte/python-can/pull/1366 is merged, I will close this PR. The combination of https://github.com/hardbyte/python-can/pull/1367 and https://github.com/hardbyte/python-can/pull/1366 will resolve everything I had intended from this PR.